File: backend/app/schemas/cobertura.py
"""Schemas para la matriz de cobertura calculada."""
from datetime import date

from pydantic import BaseModel


class CoberturaCell(BaseModel):
    balance: float
    color: str  # 'white' | 'green' | 'yellow' | 'red'


class CoberturaRow(BaseModel):
    producto_id: int
    sku: str
    producto: str
    unidad: str
    celdas: list[CoberturaCell]  # mismo orden que `buckets`


class CoberturaMatrix(BaseModel):
    snapshot_id: int
    granularidad: str  # 'semanal' | 'mensual'
    planta: str  # codigo de planta o 'TODAS'
    buckets: list[date]  # fecha inicio de cada bucket (semana o mes)
    filas: list[CoberturaRow]

File: frontend/src/lib/api/cobertura.ts
import { request } from './client'
import type { ColorCobertura } from '../colors'

export interface CoberturaCell {
  balance: number
  color: ColorCobertura
}

export interface CoberturaRow {
  producto_id: number
  sku: string
  producto: string
  unidad: string
  celdas: CoberturaCell[]
}

export interface CoberturaMatrix {
  snapshot_id: number
  granularidad: 'semanal' | 'mensual'
  planta: string
  buckets: string[]   // ISO dates
  filas: CoberturaRow[]
}

export interface CoberturaParams {
  granularidad?: 'semanal' | 'mensual'
  planta?: string         // 'TODAS' o código
  n_buckets?: number
}

export const coberturaApi = {
  get: (snapshotId: number, params: CoberturaParams = {}) => {
    const qs = new URLSearchParams()
    if (params.granularidad) qs.set('granularidad', params.granularidad)
    if (params.planta) qs.set('planta', params.planta)
    if (params.n_buckets) qs.set('n_buckets', String(params.n_buckets))
    const q = qs.toString()
    return request<CoberturaMatrix>(`/snapshots/${snapshotId}/cobertura${q ? '?' + q : ''}`)
  },
}
